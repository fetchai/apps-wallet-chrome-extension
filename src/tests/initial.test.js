import React from 'react'
import  ReactDOM  from 'react-dom'
import Initial from '../views/initial'
import chrome from 'sinon-chrome';
import renderer from 'react-test-renderer'
import {render, cleanup, fireEvent, within} from '@testing-library/react'
import '@testing-library/jest-dom'
import Router, { clear } from '../services/router'

describe.skip(':Initial', () => {

  beforeAll(() => {
    global.chrome = chrome;
  })

  afterEach(() => {
    cleanup();
    clear()
  })

  //
  // test('initial renders without crashing', () => {
  //   const div = document.createElement('div')
  //   ReactDOM.render(<Router><Initial/></Router>, div)
  //   debugger
  //   ReactDOM.unmountComponentAtNode(div)
  // })
  //


  test.skip('initial renders without crashing', () => {
   const container = render(<Initial />)
  expect(container).toMatchSnapshot()
  })

  test.skip('initial snapshot test', () => {
    const component = renderer.create(<Initial/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test(' create button renders correctly', () => {
    const { getByTestId } = render(<Initial/>);
    const appHeader = getByTestId('recover_button')
  })

  test('recover button renders correctly', () => {
    const { getByTestId } = render(<Initial/>);
    const button = getByTestId('recover_button')
    expect(button).toHaveTextContent("Recover")
  })

  test('clicking on recover button redirects to recover component', () => {
    const { getByTestId, getAllByTestId } = render(<Router><Initial/></Router>);
    const button = getByTestId('recover_button')
    fireEvent(
      button,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }))
    // tests the recover component is now mounted by router.
    const app = getAllByTestId('recover')
    expect(app.length).toBe(1);
  })

  test('clicking on create button redirects to create component', () => {
    const { getByTestId, getAllByTestId } = render(<Router><Initial/></Router>);
    const button = getByTestId('create_button')

    fireEvent(
      button,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }))

    // tests the create component is now mounted.
    const app = getAllByTestId('create')
    expect(app.length).toBe(1);
  })

  test.skip('try test redirection in oncomponent mount', () => {
  localStorage.setItem("x", "y")
  const x = localStorage.getItem("x")
    console.log(" x is about : " + x)
  })

})



