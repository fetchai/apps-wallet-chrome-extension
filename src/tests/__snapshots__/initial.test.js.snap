// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:Initial initial renders without crashing 1`] = `undefined`;

exports[`:Initial initial snapshot test 1`] = `
<div
  id="my-extension-root-inner"
>
  <video
    autoPlay="autoplay"
    className="vid"
    loop="loop"
    muted="muted"
    playsInline="playsinline"
  >
    <source
      type="video/mp4"
    />
  </video>
  <div
    className="overlay1"
  >
    <img
      alt="Fetch.ai's Logo"
      className="logo"
    />
  </div>
  <div
    className="overlay2"
  >
    <div
      className="overlay3"
    >
      <button
        className="button-free-standing"
        data-testid="recover_button"
        onClick={[Function]}
      >
        Recover
      </button>
      <button
        className="button-free-standing"
        onClick={[Function]}
      >
        Create
      </button>
    </div>
  </div>
</div>
`;
